src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -58,6 +58,12 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
                 return;
             }
 
+            var expressionSymbol = semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol;
+            if (expressionSymbol is IParameterSymbol)
+            {
+                return;
+            }
```

Comment:
Is the new behavior expected for this case?

```csharp
public class C
{
    public void M(string s)
    {
        localFunction();

        void localFunction()
        {
            _ = s.ToString();
        }
    }
}
```

Or do we need to tighten the condition to make sure the parameter symbol corresponds to the most inner containing method?

Reply:
Wonder if it's that particular scenario that's the motivation for making the suggestion in the first place.

Reply:
@Joe4evr The original scenario looks different to me.

Reply:
For this scenario, it should still be possible to explicitly pass the parameter instead of implicit capture. Whether that's part of the current refactoring or some other is of less importance to me.